src/python/pants/backend/go/target_types.py

```diff
@@ -15,15 +15,60 @@ class GoPackageSources(GoSources):
 class GoImportPath(StringField):
     # TODO: Infer the import path from the closest ancestor `go_module` target once that target is supported.
     alias = "import_path"
-    help = "Import path in Go code to import this package."
+    help = "Import path in Go code to import this package or module."
+
+
+class GoPackageDependencies(Dependencies):
+    pass
 
 
 class GoPackage(Target):
     alias = "go_package"
-    core_fields = (*COMMON_TARGET_FIELDS, Dependencies, GoPackageSources, GoImportPath)
+    core_fields = (*COMMON_TARGET_FIELDS, GoPackageDependencies, GoPackageSources, GoImportPath)
     help = "A single Go package."
 
 
+# `go_module` target
+
+
+class GoModuleSources(Sources):
+    default = ("go.mod", "go.sum")
+
+
+class GoModule(Target):
+    alias = "go_module"
+    core_fields = (
+        *COMMON_TARGET_FIELDS,
+        Dependencies,
+        GoModuleSources,
+    )
+
+
+# `go_external_module`
+
+
+class GoExternalModulePath(StringField):
+    alias = "path"
+
+
+class GoExternalModuleVersion(StringField):
+    alias = "version"
+
+
+class GoExternalModule(Target):
+    alias = "go_external_module"
+    core_fields = (
+        *COMMON_TARGET_FIELDS,
+        Dependencies,
+        GoExternalModulePath,
+        GoExternalModuleVersion,
+        GoImportPath,
+    )
+
+
+# `go_binary` target
```

Comment:
Consider using this style:

https://github.com/pantsbuild/pants/blob/e01e3535b998d3616d5f1213dd1eaa0346c59c9f/src/python/pants/engine/internals/graph.py#L89-L91

Reply:
There are no rules in this file. This comment style doesn't make sense for just the target type definitions. It would make sense for the files with rules in them.

Reply:
We use it for `python/target_types.py` too:

https://github.com/pantsbuild/pants/blob/e3924570d12d72106a29e2c2780ccaa36f0a9c35/src/python/pants/backend/python/target_types.py#L85-L87

The general idea is to have section headers that more clearly delineate the file, sort of like chapters in a book.

Reply:
Ah, I had thought you were referring to the `Foo -> Bar` notation and not the dashes. Yeah the dashes would be relevant to the comments here.